lazer/contracts/evm/test/PythLazer.t.sol

```diff
@@ -71,4 +73,396 @@ contract PythLazerTest is Test {
+            timestamp,
+            uint8(channel),
+            uint8(feedsData.length)
+        );
+
+        for (uint256 i = 0; i < feedsData.length; i++) {
+            payload = bytes.concat(payload, feedsData[i]);
+        }
+
+        return payload;
+    }
+
+    function buildFeedData(
+        uint32 feedId,
+        bytes[] memory properties
+    ) internal pure returns (bytes memory) {
+        return
+            abi.encodePacked(
+                feedId,
+                uint8(properties.length),
+                bytes.concat(
+                    properties[0],
+                    properties.length > 1 ? properties[1] : bytes("")
+                )
+            );
+    }
+
+    function concatProperties(
+        bytes[] memory properties
+    ) internal pure returns (bytes memory) {
+        bytes memory result = "";
+        for (uint256 i = 0; i < properties.length; i++) {
+            result = bytes.concat(result, properties[i]);
+        }
+        return result;
+    }
+
+    function buildFeedDataMulti(
+        uint32 feedId,
+        bytes[] memory properties
+    ) internal pure returns (bytes memory) {
+        bytes memory propertiesBytes = concatProperties(properties);
+        return
+            abi.encodePacked(feedId, uint8(properties.length), propertiesBytes);
+    }
+
+    /// @notice Build a property with given ID and encoded value bytes
+    /// @param propertyId The property ID (0-8)
+    /// @param valueBytes The encoded value (int64/uint64 = 8 bytes, uint16/int16 = 2 bytes)
+    function buildProperty(
+        uint8 propertyId,
+        bytes memory valueBytes
+    ) internal pure returns (bytes memory) {
+        return abi.encodePacked(propertyId, valueBytes);
+    }
+
+    function encodeInt64(int64 value) internal pure returns (bytes memory) {
+        return abi.encodePacked(value);
+    }
+
+    function encodeUint64(uint64 value) internal pure returns (bytes memory) {
+        return abi.encodePacked(value);
+    }
+
+    function encodeInt16(int16 value) internal pure returns (bytes memory) {
+        return abi.encodePacked(value);
+    }
+
+    function encodeUint16(uint16 value) internal pure returns (bytes memory) {
+        return abi.encodePacked(value);
+    }
+
+    /// @notice Test parsing single feed with all 9 properties
+    function test_parseUpdate_singleFeed_allProperties() public pure {
+        bytes[] memory properties = new bytes[](9);
+        properties[0] = buildProperty(0, encodeInt64(100000000)); // price
+        properties[1] = buildProperty(1, encodeInt64(99000000)); // bestBid
+        properties[2] = buildProperty(2, encodeInt64(101000000)); // bestAsk
+        properties[3] = buildProperty(3, encodeUint16(5)); // publisherCount
+        properties[4] = buildProperty(4, encodeInt16(-8)); // exponent
+        properties[5] = buildProperty(5, encodeInt64(50000)); // confidence
+        properties[6] = buildProperty(6, encodeInt64(123456)); // fundingRate
```

Comment:
This is not correct because there is a presence byte (as noted above).

It would be good to fetch real price and funding rate updates from Lazer staging and parse them in the tests.